File: lib_base/src/main/java/com/android/base/app/fragment/delegates/FragmentDelegateOwner.java
package com.android.base.app.fragment.delegates;


import com.github.dmstocking.optional.java.util.function.Predicate;

import androidx.annotation.UiThread;

@UiThread
public interface FragmentDelegateOwner {

    void addDelegate(FragmentDelegate fragmentDelegate);

    boolean removeDelegate(FragmentDelegate fragmentDelegate);

    FragmentDelegate findDelegate(Predicate<FragmentDelegate> predicate);

}

File: lib_foundation/src/main/java/com/android/base/foundation/fragment/FragmentDelegateOwner.kt
package com.android.base.foundation.fragment

import androidx.annotation.UiThread

@UiThread
interface FragmentDelegateOwner {

    fun addDelegate(fragmentDelegate: FragmentDelegate<*>)

    fun removeDelegate(fragmentDelegate: FragmentDelegate<*>): Boolean

    fun findDelegate(predicate: (FragmentDelegate<*>)->Boolean): FragmentDelegate<*>?

}